### PR TITLE
[SPARK-18662] Move resource managers to separate directory

### PR DESCRIPTION
**Pull request by foxish**
## What changes were proposed in this pull request?

* Moves yarn and mesos scheduler backends to resource-managers/ sub-directory (in preparation for https://issues.apache.org/jira/browse/SPARK-18278) 
* Corresponding change in top-level pom.xml.

Ref: https://github.com/apache/spark/pull/16061#issuecomment-263649340

## How was this patch tested?

* Manual tests

/cc @rxin

### CODE REVIEW

**Comment by JoshRosen** [comment]
There's a few other places that will need updating, including the `source_file_regexes` in the test runner configuration (`modules.py`): https://github.com/apache/spark/blob/0a811210f809eb5b80eae14694d484d45b48b3f6/dev/sparktestsupport/modules.py#L472 and https://github.com/apache/spark/blob/0a811210f809eb5b80eae14694d484d45b48b3f6/dev/sparktestsupport/modules.py#L488

**Comment by JoshRosen** [comment]
Jenkins, this is ok to test

**Comment by srowen** [comment]
Hm, is this really necessary? I know it's just a move but for 2-3 modules, does it matter enough?

**Comment by tnachen** [comment]
Changed suggested by @rxin

**Comment by ash211** [comment]
With a k8s backend on the way I do think it adds a nice organization for these 3 clearly grouped modules

**Comment by foxish** [comment]
@JoshRosen Thanks for reviewing.
Updated. PTAL.

**Comment by foxish** [comment]
The test failure appears to be an unrelated flake.

**Comment by vanzin** [comment]
@foxish this need to be rebased.

**Comment by foxish** [comment]
@vanzin Done. Thanks!

File: dev/sparktestsupport/modules.py [hunk 1]
```diff
@@ -469,7 +469,7 @@ def __hash__(self):
     name="yarn",
     dependencies=[],
     source_file_regexes=[
-        "yarn/",
+        "resource-managers/yarn/",
```
Comment (vanzin) [inline dev/sparktestsupport/modules.py]: You also need to update L477 (and probably a similar change in the mesos entry).

Comment (foxish) [inline dev/sparktestsupport/modules.py]: @vanzin If I understand correctly, the sbt test goals can stay the same? 
I can see them defined the same way for other modules such as `sketch` (as `sketch/test`), but it is nested under `common/`.

Comment (vanzin) [inline dev/sparktestsupport/modules.py]: Oh. Nevermind, I thought those were file paths like these. So fine the way they are.

Comment (foxish) [inline dev/sparktestsupport/modules.py]: @vanzin also rebased after recent changes to HEAD. Is this okay to merge?

**Comment by vanzin** [comment]
LGTM, merging to master.

**Comment by foxish** [comment]
Thanks @vanzin :)

**Comment by mgummelt** [comment]
@foxish @vanzin I wish you would have cc'd me on this.  If you're going to do something significant to Mesos code, please cc me.

**Comment by foxish** [comment]
Will cc you on future issues. Apologies if this disrupted your workflow. It was meant to be a quick change only in code structure.

**Pull request closed without merging**